File: k_projects/classic/collatz_conjecture/collatz_conjecture.py
# by Don24Kane

# Python

n = int(input("enter your number: "))

steps = 0

while n > 1:
    if n % 2 == 0:
        n //= 2
    else:
        n = n * 3 + 1
    steps += 1

print("the number of steps it took for n to reach the value of 1 is:", steps)

File: k_projects/classic/collatz_conjecture/collatz_conjecture.go
# by Don24Kane

# GOlang

package main

import (
	"fmt"
)

func main() {
	var n int
	fmt.Print("enter your number: ")
	fmt.Scan(&n)

	steps := 0

	for n > 1 {
		if n%2 == 0 {
			n /= 2
		} else {
			n = n*3 + 1
		}
		steps++
	}

	fmt.Printf("the number of steps it took for n to reach the value of 1 is: %d\n", steps)
}
